packages/react/test/specs/components/Checkbox/CheckboxBase-test.tsx

```diff
@@ -0,0 +1,185 @@
+import * as React from 'react'
+import { mount } from 'enzyme'
+import CheckboxBase from 'src/components/Checkbox/CheckboxBase'
+
+describe("Baby's first test", () => {
+  it('renders something', () => {
+    expect(mount(<CheckboxBase />)).toBeTruthy()
+  })
+
+  it('has an input', () => {
+    const control = mount(<CheckboxBase />)
+    const input = control
+      .find('input')
+      .first()
+      .getDOMNode()
+    expect(input).toBeTruthy()
+  })
+
+  it('can be checked', () => {
+    const control = mount(<CheckboxBase checked={true} />)
+    const inputChecked = control
+      .find('input')
+      .first()
+      .prop('checked')
+    expect(inputChecked).toBe(true)
+  })
+
+  it('can be unchecked', () => {
+    const control = mount(<CheckboxBase />)
+    const inputChecked = control
+      .find('input')
+      .first()
+      .prop('checked')
+    expect(inputChecked).toBe(false)
+  })
+
+  it('can switch check state', () => {
+    const control = mount(<CheckboxBase />)
+    const inputChecked = control
+      .find('input')
+      .first()
+      .prop('checked')
+    expect(inputChecked).toBe(false)
+    control.setProps({ checked: true })
+    const inputCheckedAfter = control
+      .find('input')
+      .first()
+      .prop('checked')
+    expect(inputCheckedAfter).toBe(true)
+  })
+
+  it('changes state when clicked', () => {
+    const control = mount(<CheckboxBase />)
+    expect(
+      control
+        .find('input')
+        .first()
+        .prop('checked'),
+    ).toBe(false)
+    control.simulate('click')
+    expect(
+      control
+        .find('input')
+        .first()
+        .prop('checked'),
+    ).toBe(true)
+    control.simulate('click')
+    expect(
+      control
+        .find('input')
+        .first()
+        .prop('checked'),
+    ).toBe(false)
+  })
+
+  it('calls onChange when checked state changes', () => {
+    const change = jest.fn()
+    const control = mount(<CheckboxBase onChange={change} />) // let, the only constant is change
```

Comment:
What's with the comment here? I don't understand it very well.

Reply:
Event handlers are tested as part of the baseline `isConfirmant()` test.  This test checks every synthetic event type and asserts more things:

```jsx
import { isConformant, handlesAccessibility } from 'test/specs/commonTests'

describe('CheckboxBase', () => {
  isConformant(CheckboxBase)
})
```

See `.github/test-a-feature.md` for more.